src/kudu/util/thread.h

```diff
@@ -368,6 +368,47 @@ class Thread : public RefCountedThreadSafe<Thread> {
 // the given entity. If 'web' is NULL, does not register the path handler.
 Status StartThreadInstrumentation(const scoped_refptr<MetricEntity>& server_metrics,
                                   WebCallbackRegistry* web);
+
+// Container class for any details we want to capture about a thread
+// TODO: Add start-time.
+// TODO: Track fragment ID.
+// fb: Move to header file, so can be used by plugin
+class ThreadDescriptor {
```

Comment:
ThreadDescriptor name is a little overloaded. May be use 'ThreadInfo' or something similar?

Reply:
This is a kudu class originally, I moved it from the cc file to header file, so that we can reuse the data structure in plugin. I think it is better to keep it to minimize the changes to kudu.